pkgs/development/haskell-modules/configuration-ghcjs.nix

```diff
@@ -18,6 +18,8 @@ self: super:
   inherit (self.ghc.bootPkgs)
     jailbreak-cabal alex happy gtk2hs-buildtools rehoo hoogle;
 
+  llvmPackages = null;
```

Comment:
Could you add a comment here for why this is necessary?  It is not immediately obvious to me why this would be necessary, or semantically why it would be needed.

Reply:
Added a comment around it. AFAIU `llvmPackages` is inherited from `configuration-ghc-8.10.x.nix` (shared across `ghcjs` and upstream `8.10`) which defined it as `llvmPackages = pkgs.lib.dontRecurseIntoAttrs self.ghc.llvmPackages;`. As `ghcjs` does not define any `llvmPackages` that makes `llvmPackages` (and packages that depend on it) attribute in this file unevaluatable without the change to recover from an error.

The change fixed `llvmPackages` evaluation into a `null` and unbreaks evaluation of haskell dependent packages. They will likely still fail to build.

Reply:
It may be nicer to `passthru.llvmPackages = null` in ghcjs?

Reply:
AFAIU that would still fail the evaluation:

```
nix-repl> pkgs.lib.dontRecurseIntoAttrs null
error:
       … in the left operand of the update (//) operator

         at /home/slyfox/dev/git/nixpkgs-haskell-updates/lib/attrsets.nix:1180:11:

         1179|     attrs:
         1180|     attrs // { recurseForDerivations = false; };
             |           ^
         1181|

       error: value is null while a set was expected
       ```
       
       Would it also need a tweak of `configuration-ghc-8.10.x.nix`?

Reply:
I guess… or tweak `lib.dontRecurseIntoAttrs`…

Reply:
Added `llvmPackages = null` to `ghcjs` and allowed `recurseIntoAttrs` / `dontRecurseIntoAttrs` to accept `null`.

Reply:
How about `llvmPackages = {}`?

Reply:
There is code that explicitly checks if `llvmPackages` is null, so that won't work. I don't care how this is solved, but `llvmPackages = {}` won't work.

Reply:
Switched `ghcjs` to `llvmPackages = null;` and added a check to `configuration-ghc-8.10.x.nix` as:

```nix
  # ghcjs does not use `llvmPackages` and exposes `null` attribute.
  llvmPackages =
    if self.ghc.llvmPackages != null
    then pkgs.lib.dontRecurseIntoAttrs self.ghc.llvmPackages
    else null;
```